ui/src/SoundWave.js

```diff
@@ -0,0 +1,70 @@
+export class SoundWave {
+
+    constructor(canvas, stream) {
+        this.canvas = canvas;
+        this.stream = stream;
+        this.analyser = undefined;
+        this.stopId = undefined;
+        this.freqs = undefined;
+        this.ctx = undefined;
```

Comment:
nit: what does ctx mean? Not entirely sure what this is representing--maybe change